peekingduck/commands/nodes.py

```diff
@@ -0,0 +1,104 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""PeekingDuck CLI `nodes` command."""
+
+import math
+from pathlib import Path
+from typing import Tuple
+
+import click
+
+from peekingduck.declarative_loader import PEEKINGDUCK_NODE_TYPES
+
+
+@click.command()
+@click.argument("type_name", required=False)
+def nodes(type_name: str = None) -> None:
+    """Lists available nodes in PeekingDuck. When no argument is given, all
+    available nodes will be listed. When the node type is given as an argument,
+    all available nodes in the specified node type will be listed.
+
+    Args:
+        type_name (str): input, augment, model, draw, dabble, or output.
+    """
+    config_dir = Path(__file__).resolve().parents[1] / "configs"
+
+    if type_name is None:
+        node_types = PEEKINGDUCK_NODE_TYPES
+    else:
+        node_types = [type_name]
+
+    for node_type in node_types:
+        click.secho("\nPeekingDuck has the following ", bold=True, nl=False)
+        click.secho(f"{node_type} ", fg="red", bold=True, nl=False)
+        click.secho("nodes:", bold=True)
+
+        node_names = [path.stem for path in (config_dir / node_type).glob("*.yml")]
+        max_length = _len_enumerate(max(enumerate(node_names), key=_len_enumerate))
+        for num, node_name in enumerate(node_names):
+            idx = num + 1
+            url = _get_node_url(node_type, node_name)
+            node_name_width = max_length + 1 - _num_digits(idx)
+
+            click.secho(f"{idx}:", nl=False)
+            click.secho(f"{node_name: <{node_name_width}}", bold=True, nl=False)
+            click.secho("Info: ", fg="yellow", nl=False)
+            click.secho(url)
+
+    click.secho("\n")
+
+
+def _get_node_url(node_type: str, node_name: str) -> str:
+    """Constructs the URL to documentation of the specified node.
+
+    Args:
+        node_type (str): One of input, model, dabble, draw or output.
+        node_name (str): Name of the node.
+
+    Returns:
+        (str): Full URL to the documentation of the specified node.
+    """
+    node_path = f"{node_type}.{node_name}"
+    url_prefix = "https://peekingduck.readthedocs.io/en/stable/nodes/"
+    url_postfix = ".html#module-"
+
+    return f"{url_prefix}{node_path}{url_postfix}{node_path}"
+
+
+def _len_enumerate(item: Tuple) -> int:
+    """Calculates the string length of an enumerate item while accounting for
+    the number of digits of the index.
+
+    Args:
+        item (Tuple): An item return by ``enumerate``, contains
+            ``(index, element)`` where ``element`` is a ``str``.
+
+    Returns:
+        (int): Sum of length of ``element`` and the number of digits of its
+            index.
+    """
+    return _num_digits(item[0] + 1) + len(item[1])
```

Comment:
The `_num_digits(item[0] + 1)` looks like it is meant to be `_num_digits(item[0]) + 1`, since current `_num_digits()` implementation counts from `0` to `n-1` (see related comment below).

Reply:
Removed the erroneous `+ 1`, it was originally used to avoid the maths error of log(0). This is changed to be handled in `_num_digits()` instead.